RestartManager.js

```diff
@@ -26,6 +28,12 @@ const RestartManager = (() => {
 
     function restartApp(onlyIfUpdateIsPending = false) {
         if (_allowed) {
+            if (_inProgress) {
+                log("A restart request is already in progress or queued");
+                return;
+            }
+            // The restart won't execute if `onlyIfUpdateIsPending === true` and there is no pending update.
+            _inProgress = !onlyIfUpdateIsPending || !!(NativeCodePush.getUpdateMetadata(CodePush.UpdateState.PENDING));
```

Comment:
Isn't this duplicating logic between this JS method and its corresponding native implementation? It might be better to have the native `restartApp` method return a `Promise` (or something similar) which allows the JS-side to determine whether the restart actually happened, at which point, you can reset the `_inProgress` variable to `false`.